src/value.rs

```diff
@@ -72,13 +72,7 @@ where
     T: Into<Value>,
 {
     fn from(values: HashMap<String, T>) -> Self {
-        let mut r = HashMap::new();
-
-        for (k, v) in values {
-            r.insert(k.clone(), v.into());
-        }
-
-        ValueKind::Table(r)
+        ValueKind::Table(values.into_iter().map(|(k, v)| (k, v.into())).collect())
```

Comment:
The new version is surely much more concise! 

But on the other hand, it is harder to understand (at least for me). Maybe for the sake of readability, it could be made into two-liner like

```rust
let map = HashMap::from_iter(...);
ValueKind::Table(map)
```

And to be sure that strings are reused [`drain`](https://doc.rust-lang.org/std/collections/hash_map/struct.HashMap.html#method.drain) could be used since function takes the ownership of `values` either way? Possibly this is what `into_iter` does, but I am not sure.

Reply:
Thanks for that comment. I will push a fixup commit for splitting the line into two shortly.

Using `HashMap::drain` is not what we want here, because it keeps the hashmap allocated for later use. `HashMap::into_iter()` takes ownership and with `collect()`ing it into a new `HashMap` object of `HashMap<String, Value>`. That's exactly what we want here, so that part is fine.

Thanks again for your comment!